library/src/main/scala/scala/scalajs/js/JSConverters.scala

```diff
@@ -51,4 +62,35 @@ object JSConverters {
       arr: scala.Array[T]): JSRichGenTraversableOnce[T] =
     new JSRichGenTraversableOnce(arr)
 
+  @inline
+  implicit def JSRichFutureThenable[A](f: Future[Thenable[A]]): JSRichFuture[A] =
+    new JSRichFuture[A](f.asInstanceOf[Future[A | Thenable[A]]])
```

Comment:
What if `A` itself is a `Thenable`? I thought JS promises flatten all the way...